premium/usage.go

```diff
@@ -321,23 +384,24 @@ func (u *BatchUpdater) backgroundUpdater() {
 	<-started
 }
 
-func (u *BatchUpdater) updateUsageWithRetryAndBackoff(ctx context.Context, numberToUpdate uint32) error {
+func (u *BatchUpdater) updateUsageWithRetryAndBackoff(ctx context.Context, rowsToUpdate uint32, tables []cqapi.UsageIncreaseTablesInner) error {
 	for retry := 0; retry < u.maxRetries; retry++ {
-		u.logger.Debug().Str("url", u.url).Int("try", retry).Int("max_retries", u.maxRetries).Uint32("rows", numberToUpdate).Msg("updating usage")
+		u.logger.Debug().Str("url", u.url).Int("try", retry).Int("max_retries", u.maxRetries).Uint32("rows", rowsToUpdate).Msg("updating usage")
 		queryStartTime := time.Now()
 
 		resp, err := u.apiClient.IncreaseTeamPluginUsageWithResponse(ctx, u.teamName, cqapi.IncreaseTeamPluginUsageJSONRequestBody{
 			RequestId:  uuid.New(),
 			PluginTeam: u.pluginMeta.Team,
 			PluginKind: u.pluginMeta.Kind,
 			PluginName: u.pluginMeta.Name,
-			Rows:       int(numberToUpdate),
+			Rows:       int(rowsToUpdate),
+			Tables:     &tables,
```

Comment:
We need to be mindful here that if the `tables` parameter is supplied, its total number of rows should equal the number given in the `rows` parameter. 

My reading of the code here is that this will not necessarily be the case, as if no table breakdowns were supplied, we will pass an empty array here, rather than nil. I haven't tested how the backend handles this, but probably we should explicitly not pass `tables` if it hasn't been populated

Reply:
Yeah I can do that, it makes sense.

Reply:
Done